Week2/Homework/MySQL exercises/aggreFunctions.js

```diff
@@ -0,0 +1,86 @@
+const util = require('util');
+const mysql = require('mysql');
+const fs=require('fs');
+const config=JSON.parse(fs.readFileSync("config_secret.json"))
+
+const connection = mysql.createConnection({
+  host: config.host,
+  user: config.user,
+  password: config.password,
+  database: config.database
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+
+async function seedDatabase() {
+ 
+    const  researchPapers_and_Thier_authors = `
+    SELECT R.paper_id,paper_title,count(A.author_id) FROM research_papers R 
+    LEFT JOIN author_research_papers A
+    ON A.paper_id = R.paper_id
+    group by A.paper_id; ` ;
+
+    const sumOf_female_researcher = `
+    SELECT count(*) AS female_researcher FROM authors A
+    JOIN author_research_papers R
+    ON A.author_no = R.author_id
+    WHERE gender='f' ;`;
+
+    const averageOf_H_index_perUnive= `
+    SELECT university,AVG(h_index)
+    FROM authors
+    GROUP BY university;
+    `
+    const sumOf_researchPapers_perUnive= `
+    SELECT university,count(R.paper_id) AS num_research_papers FROM authors A JOIN author_research_papers R
+    ON A.author_no = R.author_id
+    GROUP BY university;
```

Comment:
It's easier to read when you put `FROM` and every single `JOIN` on a separate line.

```suggestion
    SELECT university, COUNT(r.paper_id) AS num_research_papers
    FROM authors a
    JOIN author_research_papers r ON a.author_no = r.author_id
    GROUP BY university;
```